docs/guide/api-hmr.md

```diff
@@ -222,6 +222,43 @@ If called before connected, the data will be buffered and sent once the connecti
 
 See [Client-server Communication](/guide/api-plugin.html#client-server-communication) for more details.
 
+## Typescript for Custom Events
+
+Internally, vite infers the type of a payload from the `CustomEventMap` interface, it is possible to type custom events by extending the interface:
+
+:::tip Note
+Make sure to include the `.d.ts` extension when specifying TypeScript declaration files. Vite exposes its exports as `/types/*` (as seen in its [package.json](https://github.com/vitejs/vite/blob/main/packages/vite/package.json#L40C1-L42)), so without the extension, the library may not know which file the module is trying to extend.
```

Comment:
```suggestion
Make sure to include the `.d.ts` extension when specifying TypeScript declaration files. Otherwise, TypeScript may not know which file the module is trying to extend.
```
I think we can skip the reason here to simplify the document.